README.md

```diff
@@ -1,35 +1,254 @@
+    <Version>1.2.3</Version>
+    <FileVersion>0.1.0</FileVersion>
+</Project>
 ```
 
-## TypeScript Linting
+<div align="left">
+<a href="#examples">More Examples Below!! 👇🏼</a>
+</div>
+
+---
+
+<div align="center">
+
+## **Action Inputs**
+</div>
+
+| Input Name | Description | Required | Default Value |
+|---|:----|:---:|---|
+| `repo-owner` | The owner of the repository | yes | N/A |
+| `repo-name` | The name of the repository | yes | N/A |
+| `branch-name` | The name of the branch where the file lives. | yes | N/A |
+| `file-format` | The type of file that contains the version. Currently, the only supported value is `xml`. | yes | N/A |
+| `file-path` | The path to the file relative to the root of the repository. | yes | N/A |
+| `version-keys` | The key(s) that can hold the version in the file. | yes | N/A |
+| `case-sensitive-keys` | If true, key searching will be case-sensitive. | no | `true` |
+| `fail-on-key-value-mismatch` | If true, will fail the action if all of the key values listed in the `version-keys` input do not match.  Other failure inputs will not affect this input. | no | `false` |
+| `fail-when-version-not-found` | If true, will fail the action if no version exists.   Other failure inputs will not affect this input. | no | `true` |
+
+---
 
-The typescript linting that is setup for this project is very explicit.  The reason for this is to keep the code base clean and to avoid any pitfalls of the language.  If any unique situations come up that violate the linting rules, they can be disabled/enabled with the comments below.  
+<div align="center" style="font-weight:bold">
 
-Disables a rule for a file.
-```js
-/* eslint-disable <typescript-rule-name> */
+## **Examples**
+</div>
+
+<div align="center">
+
+### **Example 1 - (Pass If Version Not Found)**
+</div>
+
+Requirements:
+- Search for a version but do not fail the workflow if no version is found
+
+⚠️The action input `fail-when-version-not-found` is not required and has a default value of `true`.  If you do not want the action to fail whe the version is not found, you must explicitly use the input.
+
+``` yml
+#Example 1 Workflow
+- name: Get Version From C# Project File
+    uses: KinsonDigital/VersionMiner@v1.0.0-preview.1
+    with:
+        repo-owner: JohnDoe
+        repo-name: MyRepo
+        branch-name: master
+        file-type: xml
+        file-path: MyProject/MyProject.csproj
+        version-keys: Version
+        fail-when-version-not-found: false
+```
+``` xml
+<!--Example 1 C# Project File-->
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <LangVersion>10.0</LangVersion>
+    <Version></Version>👈🏼No value.  Does not fail workflow.
+</Project>
+```
+
+<div align="center">
+
+### **Example 2 - (Multiple Key Searching)**
+</div>
+
+Requirements:
+- Search multiple keys for the version.
+- Fail the workflow if no version are found in the keys.
```

Comment:
Question.

Should I use this?
```suggestion
- Fail the workflow if no versions are found in the keys.
```

The reason I ask is because you can use the action to check for a single version in multiple keys in the file.

Example:
``` yaml
- name: Get Version From C# Project File
    uses: KinsonDigital/VersionMiner@v1.0.0-preview.1
    with:
        repo-owner: JohnDoe
        repo-name: MyRepo
        branch-name: master
        file-type: xml
        file-path: MyProject/MyProject.csproj
        version-keys: Version,FileVersion 👈🏼 This guy here
```

The `version-keys` input can have a comma delimited list of keys that a version **MIGHT** exist in.
So in the example above, the key `Version` AND `FileVersion` are checked to see if they contain a version value.
This means that one of them could have a version or BOTH of them could have a version.  Only one of them need a version though.

I didn't know if the word **version** needed to be plural which means we would still use the word **are**